settingshelper.py

```diff
@@ -70,39 +70,39 @@ def get_temp_file(self, suffix="", prefix="tmp"):
         return os.path.join(self.temp_dir, name)
 
 
-def get_server_url(http_method, server_root, username, password):
+def get_couchdb_url(server_root, username, password, dbname, use_https=False):
     if username and password:
-        return '%(http_method)s://%(user)s:%(pass)s@%(server)s' % {
-            'http_method': http_method,
-            'user': username,
-            'pass': password,
-            'server': server_root,
+        db_params = {
+            "username": username,
+            "auth_token": password,
+            "use_basic_auth": True,
         }
     else:
-        return '%(http_method)s://%(server)s' % {
-            'http_method': http_method,
-            'server': server_root,
-        }
+        # admin party + empty username and auth_token
+        db_params = None
+    http_method = 'https' if use_https else 'http'
+    uri = '{http_method}://{server_root}/{dbname}'.format(**locals())
```

Comment:
no sure about this pattern of using `locals()`. Seem like it could lead to bugs down time line if someone renames somehting.

Reply:
[Python 3 has syntax for this](https://www.python.org/dev/peps/pep-0498/). I think it will become a common pattern once we switch.